connectors/sources/salesforce.py

```diff
@@ -80,58 +110,66 @@ async def close(self):
         retries=RETRIES,
         interval=RETRY_INTERVAL,
     )
-    async def get_token(self, enable_refresh=True):
-        # TODO add refresh method
-        self._logger.debug("Fetching Salesforce token...")
-        resp = await self.session.post(
-            f"{self.base_url}{TOKEN_ENDPOINT}", data=self.token_payload
-        )
-        resp_json = await resp.json()
-        self.token = resp_json["access_token"]
-        self.token_issued_at = resp_json["issued_at"]
-        self.token_refresh_enabled = enable_refresh
-        self._logger.debug("Salesforce token retrieved.")
+    async def get_token(self):
```

Comment:
I'd probably move token logic into separate class.

Also, it might make sense to wrap the logic in a mutex.

That also gave me a hint, that we're missing this logic in Sharepoint Online connector!

Reply:
mutex + separate class sounds really clean, I like it! Thanks!